src/data/search.ts

```diff
@@ -35,13 +37,30 @@ export type ItemType =
   | "automation_blueprint"
   | "script_blueprint";
 
-export const findRelated = (
+export const findRelated = async (
   hass: HomeAssistant,
   itemType: ItemType,
   itemId: string
-): Promise<RelatedResult> =>
-  hass.callWS({
+): Promise<RelatedResult> => {
+  const related = await hass.callWS<RelatedResult>({
     type: "search/related",
     item_type: itemType,
     item_id: itemId,
   });
+
+  Object.keys(related).forEach((key) => {
+    related[key] = related[key]
+      .map((id: string) => hass.states[id])
```

Comment:
Not all related items are entities, so this does not makes sense. Only for scenes, scripts, automations and entity it makes sense.